src/java.base/share/classes/java/lang/ThreadGroup.java

```diff
@@ -185,7 +185,14 @@ public final int getMaxPriority() {
      * @return  {@code true} if this thread group is a daemon thread group;
      *          {@code false} otherwise.
      * @since   1.0
+     *
+     * @deprecated Destroying a thread group and its subgroups is not an atomic
+     *             operation and is inherently racy with respect to threads that
+     *             have been created but have not started. The concept of daemon
+     *             thread group that is automatically destroyed will be removed
+     *             in a future release.
      */
```

Comment:
I would switch the order of the sentences.  Put the action first and the rationale second. $.02
(in all the @ deprecated javadoc).